### PR TITLE
add CAPD integration tests for telemetry

### PR DESCRIPTION
**Pull request by maralavi**
**What this PR does / why we need it:**
- This PR is to add CAPD integration tests for telemetry

**Describe testing done for PR:**
Added integration tests as part of the PR

**Does this PR introduce a user-facing change?:**
None

**Release note**:

```release-note
added CAPD integration tests for telemetry
```

**New PR Checklist**

- [X] Ensure PR contains only public links or terms
- [X] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [X] Squash the commits in this branch before merge to preserve our git history
- [ ] If this PR is just an idea or POC, use a [Draft PR](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests) instead of a full PR
- [X] Add appropriate [kind label](../docs/release/kind-labels.md) according to what type of issue is being addressed.

### CODE REVIEW

File: pkg/v1/tkg/test/tkgctl/shared/ceip.go [hunk 1]
```diff
@@ -118,3 +133,83 @@ func verifyTelemetryJobURL(context context.Context, url string, mcProxy *framewo
...
+		return true, nil
+	}); err != nil {
+		return err
+	}
+	if len(pods.Items) == 0 {
+		return errors.New("no telemetry pod is running")
+	}
+
+	// check to make sure that the telemetry pod does not have "Failed" status
+	if pods.Items[0].Status.Phase == "Failed" {
```
Comment (vijaykatam) [inline pkg/v1/tkg/test/tkgctl/shared/ceip.go:200]: I would check if the pod phase is running and this should use a wait with poll or preferably gomega's `Eventually` assertion.

File: pkg/v1/tkg/test/tkgctl/docker/docker_ceip_test.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// nolint:typecheck,nolintlint
```
Comment (vijaykatam) [inline pkg/v1/tkg/test/tkgctl/docker/docker_ceip_test.go:4]: (question): why do we need these disabled?

File: pkg/v1/tkg/test/tkgctl/shared/ceip.go (hunk 1 above)
Comment (danniel1205) [inline pkg/v1/tkg/test/tkgctl/shared/ceip.go:200]: question: do we want to check the status for all pods ?

File: pkg/v1/tkg/test/tkgctl/shared/ceip.go [hunk 3]
```diff
@@ -118,3 +133,83 @@ func verifyTelemetryJobURL(context context.Context, url string, mcProxy *framewo
...
+
+	scheme := mcProxy.GetScheme()
+	batchv1.AddToScheme(scheme)
+
+	cronJob := &v1beta1.CronJob{}
+	if err = client.Get(context, types.NamespacedName{Name: telemetryName, Namespace: telemetryNamespace}, cronJob); err != nil {
+		return err
+	}
+	// updating the telemetry cron job schedule to "* * * * *" so that the cronjob can be scheduled to run within the next 59 seconds
+	cronJob.Spec.Schedule = "* * * * *"
```
Comment (danniel1205) [inline pkg/v1/tkg/test/tkgctl/shared/ceip.go:154]: Suggestion: please deepCopy and make changes to the copied object instead. It is risky to change the pointer directly. The same to L205 as well.

File: pkg/v1/tkg/test/tkgctl/docker/docker_ceip_test.go (hunk 2 above)
Comment (danniel1205) [reply pkg/v1/tkg/test/tkgctl/docker/docker_ceip_test.go:4]: had the same question.

Comment (maralavi) [reply pkg/v1/tkg/test/tkgctl/docker/docker_ceip_test.go:4]: I just added this file similar to other test files under tkgctl/docker added by the CLI team. I'll check what would happen with these removed and will let you know in a bit

File: pkg/v1/tkg/test/tkgctl/shared/ceip.go (hunk 1 above)
Comment (maralavi) [reply pkg/v1/tkg/test/tkgctl/shared/ceip.go:200]: @danniel1205 The cluster and cronjob is created just then, and so the job is created just then and there is only one pod running, as I return the schedule to create telemetry job to once every six hours right after the test.

File: pkg/v1/tkg/test/tkgctl/shared/ceip.go (hunk 3 above)
Comment (maralavi) [reply pkg/v1/tkg/test/tkgctl/shared/ceip.go:154]: Thanks for this. Will fix

File: pkg/v1/tkg/test/tkgctl/shared/ceip.go (hunk 1 above)
Comment (maralavi) [reply pkg/v1/tkg/test/tkgctl/shared/ceip.go:200]: Sure, will change to only allow either of  'running' or 'succeeded' statuses

**Review by vijaykatam — APPROVED** [review]
LGTM

**Review by danniel1205 — APPROVED** [review]

**Pull request merged**